src/lib/common/writer/content-entry.ts

```diff
@@ -24,3 +36,253 @@ export const writeContentLocale = async (locale: InferInsertModel<typeof content
 
   return dbRes;
 };
+
+export const writeContentEntry = async (
+  contentEntryInput: ContentEntry,
+  blueprintMaps: BlueprintPaginationResult,
+  documentFieldMap: typeof DocumentFieldMap,
+) => {
+  const contentEntryResult = await insertContentEntry(contentEntryInput);
+  await insertContentEntryFields(
+    contentEntryInput.fields,
+    contentEntryInput.documentId,
+    blueprintMaps,
+    1,
+    documentFieldMap,
+  );
+  return contentEntryResult;
+};
+
+export const writePublishedContentEntryFields = async (
+  contentEntryInput: ContentEntry,
+  blueprintMaps: BlueprintPaginationResult,
+  documentFieldMap: typeof DocumentFieldMap,
+) => {
+  await areDocumentFieldsMatching(contentEntryInput, documentFieldMap, blueprintMaps);
+  return;
+};
+
+const insertContentEntry = async (contentEntryInput: ContentEntry) => {
+  try {
+    return await db
+      .insert(contentEntry)
+      .values({
+        id: contentEntryInput.documentId,
+        title: contentEntryInput.title,
+        status: contentEntryInput.status,
+        contentTypeId: contentEntryInput.blueprintId,
+        contentTypeVariant: contentEntryInput.blueprintVariant,
+        previewImageUrl: contentEntryInput.previewImageUrl,
+      })
+      .returning({
+        id: contentEntry.id,
+        title: contentEntry.title,
+        status: contentEntry.status,
+        contentTypeId: contentEntry.contentTypeId,
+        contentTypeVariant: contentEntry.contentTypeVariant,
+        previewImageUrl: contentEntry.previewImageUrl,
+      })
+      .execute();
+  } catch (err) {
+    console.log(` insertContentEntry`);
+    throw new Error(err);
+  }
+};
+
+// export const writeContentEntryFields = async (
+//   contentEntryFieldsInput: InferInsertModel<typeof contentEntryField>[],
+// ) => {
+//   const contentEntryFieldsResult = await insertContentEntryFields(contentEntryFieldsInput);
+//   return contentEntryFieldsResult;
+// };
+
+const insertContentEntryFields = async (
```

Comment:
this looks overkill to me @gkoutzioulis 

for me it can all run in one sqlite query - maybe you craft a native sqlite query first test in tableplus and then we can see togehter how to make it work in drizzle